packages/@lwc/synthetic-shadow/src/faux-shadow/shadow-token.ts

```diff
@@ -44,3 +49,30 @@ defineProperty(Element.prototype, KEY__SHADOW_TOKEN, {
     },
     configurable: true,
 });
+
+defineProperty(Element.prototype, KEY__SHADOW_STATIC, {
+    set(this: Element, v: boolean) {
+        // Marking an element as static will propagate the shadow resolver to the children.
+        const fn = (this as any)[KEY__SHADOW_RESOLVER];
+        if (v) {
+            const treeWalker = createTreeWalker.call(
+                getOwnerDocument(this),
+                this,
+                NodeFilter.SHOW_ELEMENT | NodeFilter.SHOW_COMMENT | NodeFilter.SHOW_TEXT,
+                () => NodeFilter.FILTER_ACCEPT,
+                false
+            );
+
+            let currentNode: Node | null;
+
+            while ((currentNode = treeWalker.nextNode())) {
```

Comment:
I don't remember if the walking is root inclusive or not... it seems that you're assigning the resolver above in `mountStatic` for the root element of the fragment, and then this goes inside... please confirm, otherwise you're doing it twice for the root.

Reply:
once you create the treeWalker, `treeWalker.currentNode` is the root, but we are checking directly for `.nextNode` to skip the root.